crypto/fipsmodule/ec/ec_key.c

```diff
@@ -329,92 +329,77 @@ int EC_KEY_check_key(const EC_KEY *eckey) {
   return 1;
 }
 
-static int EVP_EC_KEY_check_fips(EC_KEY *key) {
-  uint8_t msg[16] = {0};
-  size_t msg_len = 16;
+static int ec_key_coordinates_range_check(const EC_KEY *key) {
+  // Check that the coordinates of the public key are within the range [0, p-1].
+  // If the public key EC point is affine we simply take the x and y
+  // coordinates, otherwise, we need to convert the point to affine first.
   int ret = 0;
-  uint8_t* sig_der = NULL;
-  EVP_PKEY *evp_pkey = EVP_PKEY_new();
-  EVP_MD_CTX *ctx = EVP_MD_CTX_new();
-  const EVP_MD *hash = EVP_sha256();
-  size_t sign_len;
-  if (!evp_pkey ||
-      !ctx ||
-      !EVP_PKEY_set1_EC_KEY(evp_pkey, key) ||
-      !EVP_DigestSignInit(ctx, NULL, hash, NULL, evp_pkey) ||
-      !EVP_DigestSign(ctx, NULL, &sign_len, msg, msg_len)) {
-    goto err;
-  }
-  sig_der = OPENSSL_malloc(sign_len);
-  if (!sig_der ||
-      !EVP_DigestSign(ctx, sig_der, &sign_len, msg, msg_len)) {
-    goto err;
-  }
-  if (boringssl_fips_break_test("ECDSA_PWCT")) {
-    msg[0] = ~msg[0];
+ 
+  BIGNUM *x = BN_new();
+  BIGNUM *y = BN_new();
```

Comment:
For completeness sake, I would add null checks for both x,y here.  It looks like the subsequent functions (ec_felem_to_bignum & BN_free & others) handle null correctly it's still better to add the checks here.

Reply:
done